detox/src/devices/Device.test.js

```diff
@@ -111,6 +111,11 @@ describe('Device', () => {
     expect(device.name).toEqual('mock-device-name-from-driver');
   });
 
+  it('should return the type from the configuration', async () => {
+    const device = validDevice();
```

Comment:
Good point. With this implementation it will return the relative-path as-is, which might be weird, but that wouldn't be the driver's fault (rather, the whole concept's "fault"). This is why I suggest to keep as-is. Nevertheless, it is overrideable by the driver's implementor.